Module-12-Assignment/Question 2.txt

```diff
@@ -0,0 +1,57 @@
+Question : Optimize the following code and quote the problems associated with current code.
+trigger accountTrigger on Account (before delete, before insert, before update) 
+{
+    List<Opportunity> opptysClosedLost = [select id, name, closedate, stagename from Opportunity
+                                          where accountId IN :Trigger.newMap.keySet() and StageName='Closed - Lost'];
+    List<Opportunity> opptysClosedWon = [select id, name, closedate, stagename from Opportunity where
+                                         accountId IN :Trigger.newMap.keySet() and StageName='Closed - Won'];
+    for(Account a : Trigger.new)
+    {
+        for(Opportunity o: opptysClosedLost)
+        {
+            if(o.accountid == a.id)
+                System.debug('Do more logic here...');
+        }
+        for(Opportunity o: opptysClosedWon)
+        {
+            if(o.accountid == a.id)
+                System.debug('Do more logic here...');
+        }
+    }
+}
+
+Answer:
```

Comment:
Trigger.newMap is available only in before update trigger. This trigger wants to do some processing on opportunities linked with accounts depicted by Trigger.newMap so this trigger should execute only for before update

Reply:
Done